packages/xstate-store/src/types.ts

```diff
@@ -340,3 +346,18 @@ export type Selector<TContext, TSelected> = (context: TContext) => TSelected;
 export interface Selection<TSelected> extends Subscribable<TSelected> {
   get: () => TSelected;
 }
+
+export interface Readable<T> extends Subscribable<T> {
+  get(): T;
+}
+
+export interface Atom<T> extends Subscribable<T>, Readable<T> {
+  /** Gets the current value of the atom. */
+  get(): T;
+  /** Sets the value of the atom using a function. */
+  set(fn: (prevVal: T) => T): void;
+  /** Sets the value of the atom. */
+  set(value: T): void;
+}
+
+export interface ReadOnlyAtom<T> extends Readable<T> {}
```

Comment:
Why do we need both of those interfaces? shouldn't we just use a single `ReadOnlyAtom`?

Reply:
makes sense